src/main/java/picard/util/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,260 @@
+            GenotypeLikelihoods.GenotypeLikelihoodsAllelePair allelePair = GenotypeLikelihoods.getAllelePair(MathUtil.indexOfMin(newPLs));
+            gb.alleles(Stream.of(allelePair.alleleIndex1, allelePair.alleleIndex2).map(allelesToKeep::get).collect(Collectors.toList()));
+            // restrict AD to the new allele subset
+            if (g.hasAD()) {
+                final int[] oldAD = g.getAD();
+                final int[] newAD = IntStream.range(0, allelesToKeep.size()).map(n -> oldAD[allelesIndex[n]]).toArray();
+                gb.AD(newAD);
+            }
+            newGTs.add(gb.make());
+        }
+        return newGTs;
+    }
+
+    /**
+     * Given a list of original alleles and a subset of new alleles to retain, find the array of old PL indices that correspond
+     * to new PL indices i.e. result[7] = old PL index of genotype containing same alleles as the new genotype with PL index 7.
+     * <p>
+     * This method is written in terms f indices rather than subsetting PLs directly in order to produce output that can be
+     * recycled from sample to sample, provided that the ploidy is the same.
+     *
+     * @param originalAlleles List of original alleles
+     * @param newAlleles      New alleles -- must be a subset of {@code originalAlleles}
+     * @return old PL indices of new genotypes
+     */
+    public static int[] subsettedPLIndices(final List<Allele> originalAlleles, final List<Allele> newAlleles) {
+        final int[] result = new int[GenotypeLikelihoods.numLikelihoods(newAlleles.size(), 2)];
+
+        for (int oldPLIndex = 0; oldPLIndex < GenotypeLikelihoods.numLikelihoods(originalAlleles.size(), 2); oldPLIndex++) {
+            final GenotypeLikelihoods.GenotypeLikelihoodsAllelePair allelePairFromPLIndex = GenotypeLikelihoods.getAllelePair(oldPLIndex);
+
+            final Allele allele1 = originalAlleles.get(allelePairFromPLIndex.alleleIndex1);
+            final Allele allele2 = originalAlleles.get(allelePairFromPLIndex.alleleIndex2);
+
+            final boolean containsOnlyNewAlleles = newAlleles.contains(allele1) && newAlleles.contains(allele2);
+
+            if (containsOnlyNewAlleles) {
+                // thus we want this PL...but we need to figure out where its new index is....
+                final int newPLIndex = GenotypeLikelihoods.calculatePLindex(newAlleles.indexOf(allele1), newAlleles.indexOf(allele2));
+                result[newPLIndex] = oldPLIndex;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Checks that a boolean is true and returns the same object or throws an {@link IllegalArgumentException}
+     *
+     * @param condition any Object
+     * @param msg       the text message that would be passed to the exception thrown when {@code !condition}.
+     * @throws IllegalArgumentException if a {@code !condition}
+     */
+    public static void validateTrue(final boolean condition, final String msg) {
```

Comment:
lets make these private, if we want to make them public they should live somewhere else

Reply:
done.